scarb/src/ops/fmt.rs

```diff
@@ -127,6 +143,37 @@ fn check_file_formatting(
     }
 }
 
+pub trait Emittable {
+    fn emit(&self, ws: &Workspace<'_>, formatted: &str);
+}
+
+impl Emittable for EmitTarget {
```

Comment:
didn't go the empty struct route, 
this felt more natural, but open to challenges